File: emp_ui/lib/models/applicant.dart
class ApplicantModel {
  final int? id;
  final String fullName;
  final String email;
  final String phone;
  final String status; // APPLIED, INTERVIEW_SCHEDULED, SELECTED, REJECTED

  ApplicantModel({
    this.id,
    required this.fullName,
    required this.email,
    required this.phone,
    required this.status,
  });

  factory ApplicantModel.fromJson(Map<String, dynamic> json) {
    return ApplicantModel(
      id: json['id'],
      fullName: json['fullName'],
      email: json['email'],
      phone: json['phone'],
      status: json['status'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      "id": id,
      "fullName": fullName,
      "email": email,
      "phone": phone,
      "status": status,
    };
  }
}

File: emp-api/src/main/java/com/example/entities/Applicant.java
package com.example.entities;

import jakarta.persistence.*;

@Entity
@Table(name = "applicants")
public class Applicant {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String fullName;
    private String email;
    private String phone;

    @Enumerated(EnumType.STRING)
    private ApplicationStatus status;

    // ================= ENUM =================
    public enum ApplicationStatus {
        APPLIED,
        INTERVIEW_SCHEDULED,
        SELECTED,
        REJECTED
    }

    // ================= GETTER & SETTER =================

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getFullName() {
        return fullName;
    }

    public void setFullName(String fullName) {
        this.fullName = fullName;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getPhone() {
        return phone;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public ApplicationStatus getStatus() {
        return status;
    }

    public void setStatus(ApplicationStatus status) {
        this.status = status;
    }
}
